gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ResolveProductDependenciesTask.java

```diff
@@ -82,13 +85,15 @@ public ResolveProductDependenciesTask() {
             Project project, BaseDistributionExtension ext, Provider<Set<ProductId>> provider) {
         TaskProvider<ResolveProductDependenciesTask> depTask = project.getTasks()
                 .register("resolveProductDependencies", ResolveProductDependenciesTask.class, task -> {
+                    Provider<Configuration> configProvider = project.provider(ext::getProductDependenciesConfig);
                     task.getServiceName().set(ext.getDistributionServiceName());
                     task.getServiceGroup().set(ext.getDistributionServiceGroup());
                     task.getDeclaredProductDependencies().set(ext.getAllProductDependencies());
-                    task.setConfiguration(project.provider(ext::getProductDependenciesConfig));
+                    task.setConfiguration(configProvider);
                     task.getOptionalProductIds().set(ext.getOptionalProductDependencies());
                     task.getIgnoredProductIds().set(ext.getIgnoredProductDependencies());
                     task.getInRepoProductIds().set(provider);
+                    task.dependsOn(configProvider);
```

Comment:
Based on reviews from @CRogers I had thought that the Provider would automatically set task dependencies. Is that not the case?

Reply:
Good question.  A property/provider that is the output of task A will create a dependency if it is set to an input property of task B.  However, because this provider is not associated with a task, I do not know if the same logic applies.  I can say the behavior was different once I added the explicit dependsOn statement.

Reply:
That might be a stupid question, but what does it actually mean to depend on a provider? The provider will be evaluated before the task runs? How is that different in evaluating the provider through a task input?

Reply:
actually, the right answer to this may be that the Configuration is not listed as an Input property to the task.  See - https://github.com/palantir/sls-packaging/blob/856fba273b088d83731675895923f7f39cfb9968/gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ResolveProductDependenciesTask.java#L153-L164

While I didn't set things up this way here (predated me), I have encountered this issue before with in similar situations.  I do not know if it is still a problem of if newer versions of gradle have some way to get around it.

Reply:
More data - The task has this method to try to enforce the dependencies between the tasks as well - https://github.com/palantir/sls-packaging/blob/856fba273b088d83731675895923f7f39cfb9968/gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ResolveProductDependenciesTask.java#L106-L130

This was added with this commit - 
https://github.com/palantir/sls-packaging/commit/3649f1b38a18abc2fc180413fb577d9958789973

I didn't change the functionality of this method when I extracted the task, but it clearly is not doing what we hoped.  I have a gradle build scan where configureProductDependencies is starting after resolveProductDependencies

Reply:
just writing notes to myself - this commit was to try to avoid building jars - https://github.com/palantir/sls-packaging/commit/b855a774d28d8d7ac618f2cce5215cbbc1dfdb41#diff-ffb538d33fb6230122fe54f923c3b269363d3fd26e7f1f38c58b9750aacf9e82R136-R143

It was about a month before the one that added the otherProjectProductDependenciesTasks  method above.  The problem with all of these is that they assume they have complete knowledge of what could write dependency info into the jar manifest (i.e. the ConfigureProductDependenciesTask).  Though I still don't know why I have seen the task ordering difference noted above.

Reply:
Discussed offline and we'll move forward with this for now. If the overhead by compiling jars becomes too bad, we can look into other options.